ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +358,33 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      completeSslHandshakes();
       this.metrics.selectorIORate.inc();
     }
     long endIo = time.milliseconds();
     this.metrics.selectorIOTime.update(endIo - endSelect);
   }
 
   /**
+   * Add those Ssl connections to connected list on handshake completion
+   */
+  private void completeSslHandshakes() {
+    Iterator<SSLTransmission> sslTransIter = pendingSslHandshakes.iterator();
+    while (sslTransIter.hasNext()) {
+      Transmission sslTransmission = sslTransIter.next();
+      if (sslTransmission != null && sslTransmission.ready()) {
+        connected.add(sslTransmission.getConnectionId());
+        metrics.selectorConnectionCreated.inc();
+        Long handshakeStartTime = sslHandshakeTimer.remove(sslTransmission.getConnectionId());
+        if (handshakeStartTime != null) {
```

Comment:
again, this is unnecessary, but I think you should simply use the other suggestion of using a single data structure.